deeplearning4j-scaleout/spark/dl4j-spark-parameterserver/src/main/java/org/deeplearning4j/spark/parameterserver/iterators/MultiPdsIterator.java

```diff
@@ -0,0 +1,44 @@
+package org.deeplearning4j.spark.parameterserver.iterators;
+
+import lombok.NonNull;
+import org.apache.spark.input.PortableDataStream;
+import org.deeplearning4j.spark.parameterserver.callbacks.PortableDataStreamCallback;
+import org.deeplearning4j.spark.parameterserver.callbacks.PortableDataStreamMDSCallback;
+import org.nd4j.linalg.dataset.DataSet;
+import org.nd4j.linalg.dataset.api.MultiDataSet;
+
+import java.util.Iterator;
+import java.util.function.Consumer;
+
+/**
+ * @author raver119@gmail.com
+ */
+public class MultiPdsIterator implements Iterator<MultiDataSet> {
```

Comment:
There's small, but significant difference. Your implementation assumes that PDS might only hold MultiDataSets inside, in serialized form. And i want to get an option to pass data from PDS to DataVec, for further vectorization.